util/helm/client.go

```diff
@@ -102,29 +110,29 @@ func fileExist(filePath string) (bool, error) {
 	return true, nil
 }
 
-func (c *nativeHelmChart) ensureHelmChartRepoPath() error {
-	c.repoLock.Lock(c.repoPath)
-	defer c.repoLock.Unlock(c.repoPath)
-
-	err := os.Mkdir(c.repoPath, 0700)
-	if err != nil && !os.IsExist(err) {
+func (c *nativeHelmChart) CleanChartCache(chart string, version string) error {
+	cachePath, err := c.getCachedChartPath(chart, version)
+	if err != nil {
 		return err
 	}
-	return nil
-}
-
-func (c *nativeHelmChart) CleanChartCache(chart string, version string) error {
-	return os.RemoveAll(c.getCachedChartPath(chart, version))
+	return os.RemoveAll(cachePath)
 }
 
-func (c *nativeHelmChart) ExtractChart(chart string, version string, passCredentials bool) (string, io.Closer, error) {
-	err := c.ensureHelmChartRepoPath()
+func createTempDir() (string, error) {
+	newUUID, err := uuid.NewRandom()
 	if err != nil {
-		return "", nil, err
+		return "", err
+	}
+	tempDir := path.Join(os.TempDir(), newUUID.String())
+	if err := os.Mkdir(tempDir, 0755); err != nil {
+		return "", err
 	}
+	return tempDir, nil
```

Comment:
not sure if you had another reason but why not replacing lines 126-130 for
`return ioutil.TempDir(os.TempDir(), newUUID.String())` ?

Reply:
`ioutil.TempDir` adds a pseudo-random suffix. https://pkg.go.dev/io/ioutil#example-TempDir

Wouldn't actually hurt anything here, but it would make the dir name look even weirder. 😆